internal/controller/proposedcommit_controller.go

```diff
@@ -98,33 +106,198 @@ func (r *ProposedCommitReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
-	branchShas, err := gitOperations.GetBranchShas(ctx, []string{pc.Spec.ActiveBranch, pc.Spec.ProposedBranch})
+	err = r.calculateStatus(ctx, &pc, gitOperations)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	err = r.creatOrUpdatePullRequest(ctx, &pc)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	err = r.Status().Update(ctx, &pc)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
+
+	return ctrl.Result{
+		Requeue:      true,
+		RequeueAfter: r.Config.RequeueDuration,
+	}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *ProposedCommitReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&promoterv1alpha1.ProposedCommit{}).
+		Complete(r)
+}
+
+func (r *ProposedCommitReconciler) getGitAuthProvider(ctx context.Context, scmProvider *promoterv1alpha1.ScmProvider, secret *v1.Secret) (scms.GitOperationsProvider, error) {
+	logger := log.FromContext(ctx)
+	switch {
+	case scmProvider.Spec.Fake != nil:
+		logger.V(4).Info("Creating fake git authentication provider")
+		return fake.NewFakeGitAuthenticationProvider(scmProvider, secret), nil
+	case scmProvider.Spec.GitHub != nil:
+		logger.V(4).Info("Creating GitHub git authentication provider")
+		return github.NewGithubGitAuthenticationProvider(scmProvider, secret), nil
+	default:
+		return nil, nil
+	}
+}
+
+func (r *ProposedCommitReconciler) calculateStatus(ctx context.Context, pc *promoterv1alpha1.ProposedCommit, gitOperations *git.GitOperations) error {
+	logger := log.FromContext(ctx)
+
+	branchShas, err := gitOperations.GetBranchShas(ctx, []string{pc.Spec.ActiveBranch, pc.Spec.ProposedBranch})
+	if err != nil {
+		return err
+	}
 	logger.Info("Branch SHAs", "branchShas", branchShas)
 
 	for branch, shas := range branchShas {
 		if branch == pc.Spec.ActiveBranch {
+
 			pc.Status.Active.Hydrated.Sha = shas.Hydrated
-			commitTime, _ := gitOperations.GetShaTime(ctx, shas.Hydrated)
+			commitTime, err := gitOperations.GetShaTime(ctx, shas.Hydrated)
+			if err != nil {
+				return err
+			}
 			pc.Status.Active.Hydrated.CommitTime = commitTime
 
 			pc.Status.Active.Dry.Sha = shas.Dry
-			commitTime, _ = gitOperations.GetShaTime(ctx, shas.Dry)
+			commitTime, err = gitOperations.GetShaTime(ctx, shas.Dry)
+			if err != nil {
+				return err
+			}
 			pc.Status.Active.Dry.CommitTime = commitTime
+
+			activeCommitStatusesState := []promoterv1alpha1.ProposedCommitCommitStatusState{}
+			for _, status := range pc.Spec.ActiveCommitStatuses {
+				var csListActive promoterv1alpha1.CommitStatusList
```

Comment:
Probably should start thinking about abstracting out the fetch/calculate into its own function, both for readability and to make it easier to parallelize the network-bound stuff.